sys/flash_map/src/flash_map.c

```diff
@@ -406,6 +406,84 @@ flash_map_read_mfg(int max_areas,
     }
 }
 
+/**
+ * Determines if the specified flash area overlaps any areas in the flash map.
+ *
+ * @param area1                 The flash area to compare against the flash
+ *                                  map.
+ *
+ * @return                      True if there is an overlap; false otherwise.
+ */
+static bool
+flash_map_area_overlaps(const struct flash_area *area1)
+{
+    const struct flash_area *area2;
+    uint32_t end1;
+    uint32_t end2;
+    int i;
+
+    for (i = 0; i < flash_map_entries; i++) {
+        area2 = &flash_map[i];
+
+        if (area1->fa_device_id == area2->fa_device_id) {
+            end1 = area1->fa_off + area1->fa_size;
+            end2 = area2->fa_off + area2->fa_size;
+
+            if (end1 >= area2->fa_off && area1->fa_off < end2) {
```

Comment:
I don't think so, but I often mess things things up so I could be wrong.

`fa_off` is the first byte in the flash area.  This byte is actually inside the flash area, so there is an overlap if another area uses this byte.

`end` is *one past* the end of the area.  If an area's end is equal to another's off, the two areas are adjacent, but not overlapping.

Reply:
Wait, you're right!  Let me fix it :).

Reply:
I wander if you have unit test that catches this condition :)

Reply:
Good idea.  I will add one.

Reply:
I updated the unit test so that it tries added an area that is sandwiched between two mfg areas.  When I change the `<` to `<=` or the `>` to `>=` the test fails, so I think I got it right this time.